op-e2e/actions/upgrades/interop_fork_test.go

```diff
@@ -0,0 +1,144 @@
+	activationBlockNumber := activationBlockHeaderA.Number
+
+	require.True(t, rollupConfigA.IsInteropActivationBlock(activationBlockHeaderA.Time),
+		"Interop should be active at the first L1 inclusion block in chain A")
+	require.True(t, rollupConfigB.IsInteropActivationBlock(activationBlockHeaderB.Time),
+		"Interop should be active at the first L1 inclusion block in chain B")
+
+	activationBlock := actors.ChainA.SequencerEngine.L2Chain().GetBlockByHash(activationBlockHeaderA.Hash())
+	activationBlockTxs := activationBlock.Transactions()
+	VerifyContractsDeployedCorrectly(t, actors, activationBlockTxs, activationBlockNumber)
+
+	actors.ActBatchAllAndMine(t)
+
+	actors.ChainA.Sequencer.ActL2PipelineFull(t)
+	actors.ChainB.Sequencer.ActL2PipelineFull(t)
+
+	// The node will exhaust L1 data,
+	// it needs the supervisor to see the L1 block first,
+	// and provide it to the node.
+	for i := 0; i < 2; i++ {
+		actors.ChainA.Sequencer.ActL2EventsUntil(t, event.Is[derive.ExhaustedL1Event], 100, false)
+		actors.ChainB.Sequencer.ActL2EventsUntil(t, event.Is[derive.ExhaustedL1Event], 100, false)
+		actors.Supervisor.SignalLatestL1(t)          // supervisor will be aware of latest L1
+		actors.ChainA.Sequencer.SyncSupervisor(t)    // supervisor to react to exhaust-L1
+		actors.ChainB.Sequencer.SyncSupervisor(t)    // supervisor to react to exhaust-L1
+		actors.ChainA.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
+		actors.ChainB.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
+	}
+
+	actors.ChainA.Sequencer.ActL1HeadSignal(t)
+	actors.ChainB.Sequencer.ActL1HeadSignal(t)
+
+	actors.ChainA.Sequencer.SyncSupervisor(t) // supervisor to react to exhaust-L1
+	actors.ChainB.Sequencer.SyncSupervisor(t) // supervisor to react to exhaust-L1
+
+	actors.Supervisor.ProcessFull(t)
+
+	actors.ChainA.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
+	actors.ChainB.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
+
+	// Verify the sync status is correct
+	statusA := actors.ChainA.Sequencer.SyncStatus()
+	require.Equal(t, activationBlockNumber.Uint64(), statusA.UnsafeL2.ID().Number)
+	require.Equal(t, activationBlockNumber.Uint64(), statusA.CrossUnsafeL2.ID().Number)
+	require.Equal(t, activationBlockNumber.Uint64(), statusA.LocalSafeL2.ID().Number)
+	require.Equal(t, activationBlockNumber.Uint64(), statusA.SafeL2.Number)
+	require.Equal(t, uint64(0), statusA.FinalizedL2.Number)
+
+	statusB := actors.ChainB.Sequencer.SyncStatus()
+	require.Equal(t, activationBlockNumber.Uint64(), statusB.UnsafeL2.ID().Number)
+	require.Equal(t, activationBlockNumber.Uint64(), statusB.CrossUnsafeL2.ID().Number)
+	require.Equal(t, activationBlockNumber.Uint64(), statusB.LocalSafeL2.ID().Number)
+	require.Equal(t, activationBlockNumber.Uint64(), statusB.SafeL2.Number)
+	require.Equal(t, uint64(0), statusB.FinalizedL2.Number)
```

Comment:
Nit: might be nice to get all the way to the activation block being finalized?

Reply:
Agreed, will fix in follow up PR.